android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/MessageCenter.java

```diff
@@ -0,0 +1,80 @@
+package com.shopify.unity.buy;
+
+import android.os.Handler;
+import android.os.HandlerThread;
+
+import com.unity3d.player.UnityPlayer;
+
+import java.lang.ref.WeakReference;
+import java.util.HashMap;
+import java.util.Map;
+
+public class MessageCenter {
+    private static final HandlerThread handlerThread = new HandlerThread("UnityMessageThread");
+    private static final Handler messageHandler = new Handler(handlerThread.getLooper());
+
+    private static Map<String, WeakReference<MessageCallbacks>> callbacksInWaiting = new HashMap<>();
```

Comment:
Sorry for miscommunication but this is redundant. 
As there won't be a strong ref to anonymous  callback created in the `UnityAndroidPayFragment` like this:
```
MessageCenter.sendMessageTo(msg, receiver, new MessageCenter.MessageCallbacks() {
```

In this case this callback will be GC right away. That's said you should either rollback this changes to `Map<String, MessageCallbacks>` or make sure you keep strong ref to callback in your `UnityAndroidPayFragment`